workflow/controller/operator.go

```diff
@@ -629,6 +629,17 @@ func (woc *wfOperationCtx) processNodeRetries(node *wfv1.NodeStatus, retryStrate
 		return woc.markNodePhase(node.Name, wfv1.NodeSucceeded), true, nil
 	}
 
+	if woc.workflowDeadline != nil && time.Now().UTC().After(*woc.workflowDeadline) {
+		var message string
+		if woc.workflowDeadline.IsZero() {
```

Comment:
JSYK: The way we detect termination has changed since you opened this PR. This info is now found in `Workflow.Spec.Shutdown`